core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/columns/ColumnWithPath.kt

```diff
@@ -61,3 +64,20 @@ public interface ColumnWithPath<out T> : DataColumn<T> {
 }
 
 public val <T> ColumnWithPath<T>.depth: Int get() = path.depth()
+
+public fun ColumnWithPath(column: DataColumn<*>, path: ColumnPath): ColumnWithPath<*> =
```

Comment:
Duplicate logic to `BaseColumn<T>.addPath(path: ColumnPath)`. They should probably call each other if we want to keep both

Reply:
That's a good point, thank you. I'll call addPath from ColumnWithPath